src/Psalm/Report/SarifReport.php

```diff
@@ -0,0 +1,95 @@
+<?php
+namespace Psalm\Report;
+
+use Psalm\Internal\Json\Json;
+use function max;
+use Psalm\Config;
+use Psalm\Report;
+
+/**
+ * SARIF report format suitable for import into any SARIF compatible solution
+ *
+ * https://docs.oasis-open.org/sarif/sarif/v2.1.0/cs01/sarif-v2.1.0-cs01.html
+ */
+class SarifReport extends Report
+{
+    public function create(): string
+    {
+        $report = [
+            'version' => '2.1.0',
+            'runs' => [
+                [
+                    'tool' => [
+                        'driver' => [
+                            'name' => 'Psalm',
+                        ],
+                    ],
+                    'results' => [],
+                ],
+            ],
+        ];
+
+        foreach ($this->issues_data as $issue_data) {
+            $jsonEntry = [
+                'ruleId' => $issue_data->type,
+                'message' => [
+                    'text' => $issue_data->message,
+                ],
+                'locations' => [
+                    [
+                        'physicalLocation' => [
+                            'artifactLocation' => [
+                                'uri' => $issue_data->file_name,
+                            ],
+                            'region' => [
+                                'startLine' => $issue_data->line_from,
+                                'startColumn' => $issue_data->column_from,
+                            ],
+                        ],
+                    ]
+                ],
+            ];
+
+            if($issue_data->taint_trace != null) {
+                $jsonEntry['codeFlows'] = [
+                    [
+                        'message' => [
+                            'text' => 'Tracing the path from user input to insecure usage',
+                        ],
+                        'threadFlows' => [
+                            [
+                                'locations' => [],
+                            ],
+                        ],
+                    ]
+                ];
+
+                foreach($issue_data->taint_trace as $trace) {
+                    if(isset($trace->line_from)) {
+                        $jsonEntry['codeFlows'][0]['threadFlows'][0]['locations'][] = [
+                            'location' => [
+                                'physicalLocation' => [
+                                    'artifactLocation' => [
+                                        'uri' => $trace->file_name,
```

Comment:
I need to check if `file_name` here is just the filename or the relative path. If the latter: Fix this.

Reply:
This is the full relative path. All good.